src/compiler/utilities.ts

```diff
@@ -1904,6 +1904,80 @@ namespace ts {
         });
     }
 
+    /**
+     * Detached comment is a comment at the top of file or function body that is separated from
+     * the next statement by space.
+     */
+    export function emitDetachedComments(currentSourceFile: SourceFile, writer: EmitTextWriter,
+        writeComment: (currentSourceFile: SourceFile, writer: EmitTextWriter, comment: CommentRange, newLine: string) => void,
+        node: TextRange, newLine: string, removeComments: boolean,
+        detachedCommentsInfo: { nodePos: number; detachedCommentEndPos: number }[]) {

```

Comment:
> also please add return type. And return type and parameter type into JSDoc

I'm not sure how to do that becase I see no other `@return` or `@param` in `utilities.ts`. Can you elaborate?